exercises/transpose/examples/success-standard/src/Transpose.hs

```diff
@@ -0,0 +1,19 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Transpose (transpose) where
+
+import Data.List (dropWhileEnd)
+import Data.Maybe (isNothing)
+
+transpose :: [String] -> [String]
+transpose [] = []
+transpose rows =
+  map toString $ foldr
+    (zipWith (:) . padRight . map Just)
+    (replicate longestLength [])
+    rows
+  where
+    longestLength = maximum $ map length rows
+    padRight line = line ++ replicate (longestLength - length line) Nothing
+    toString line = concatMap (\case Nothing -> " "
```

Comment:
As evidenced by the `Just c -> c:""` arm (since only a `Char` can be `:` to a `[Char]`), the lambda is `Maybe Char -> [Char]`, but each resulting `[Char]` has exactly one element (either a single space or the one character contained in the `Just`), so instead this should be a `map` using a `Maybe Char -> Char` lambda.

```haskell
xs = [1, 2, 3]
exprA = concatMap (\x -> [x + 1]) xs
exprB = map (\x -> x + 1) xs
```

Reply:
Ah, I see. Mapping out the types is helpful in these cases.

Reply:
Since this is the case, I can probably just use `fromMaybe` and get rid of the lambda entirely.

Reply:
Yes, `fromMaybe` would be good too!